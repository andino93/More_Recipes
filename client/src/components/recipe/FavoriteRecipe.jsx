import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'react-proptypes';

/**
 * Recipe
 * @param { object } props
 * @returns { object } object
 */
const RecipeFavorite = (props) => {
  const image = (props.recipe.imageUrl !== '') ? props.recipe.imageUrl
    : process.env.DEFAULT_IMAGE;
  return (
    <div className="col-12 col-sm-6 col-md-4 col-lg-3 align-self-center hvr-float-shadow">
      <div className="col-md-12 recipe">
        <div className="recipe-img" style={{
          backgroundImage: `url(${image})`
        }} />
        <div className="recipe-meta pt-5">
          <Link to={`/recipes/${props.recipe.id}/`}>
            <h6 className="recipe-name">{props.recipe.name}</h6>
          </Link>
          <h6>
            <small>
              <i className="fa fa-tags" /> African Dishes</small>
          </h6>
          <h6 className="text-muted">
            <small>
              <i className="fa fa-user" /> {props.recipe.User.username}</small>
          </h6>
        </div>
        <div className="recipe-met2">
          <hr className="m-1" />
          <p className="text-left">
            <small>
              <span>
                <i className="fa fa-eye text-muted" /> {props.recipe.views}</span> &nbsp;
              <span>
                <i className="fa fa-thumbs-o-up text-muted" /> 211
              </span>
            </small>
          </p>
        </div>
      </div>
    </div>
  );
};

RecipeFavorite.propTypes = {
  recipe: PropTypes.object.isRequired
};

export default RecipeFavorite;
