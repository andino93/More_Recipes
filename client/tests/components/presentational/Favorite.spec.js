import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import FavoriteRecipe from '../../../src/components/recipe/FavoriteRecipe.jsx';

configure({ adapter: new Adapter() });

describe('<FavoriteRecipe>', () => {
  const store = mockStore({});
  const connectedRecipe = mountWrap(FavoriteRecipe, store, { recipe: {
    User: {
      username: 'Human'
    }
  } });
  it('should render the Recipe Component', (done) => {
    expect(toJson(connectedRecipe)).toMatchSnapshot();
    done();
  });
});
