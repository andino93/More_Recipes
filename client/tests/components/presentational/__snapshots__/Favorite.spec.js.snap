// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteRecipe> should render the Recipe Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <RecipeFavorite
        recipe={
          Object {
            "User": Object {
              "username": "Human",
            },
          }
        }
      >
        <div
          className="col-12 col-sm-6 col-md-4 col-lg-3 align-self-center hvr-float-shadow"
        >
          <div
            className="col-md-12 recipe"
          >
            <div
              className="recipe-img"
              style={
                Object {
                  "backgroundImage": "url(undefined)",
                }
              }
            />
            <div
              className="recipe-meta pt-5"
            >
              <Link
                replace={false}
                to="/recipes/undefined/"
              >
                <a
                  href="/recipes/undefined/"
                  onClick={[Function]}
                >
                  <h6
                    className="recipe-name"
                  />
                </a>
              </Link>
              <h5
                className="text-muted"
              >
                <small>
                  <i
                    className="fa fa-user"
                  />
                   
                  Human
                </small>
              </h5>
            </div>
            <div
              className="recipe-met2"
            >
              <hr
                className="m-1"
              />
              <p
                className="text-left"
              >
                <small>
                  <span>
                    <i
                      className="fa fa-eye text-muted"
                    />
                     
                  </span>
                    
                  <span>
                    <i
                      className="fa fa-thumbs-o-up text-muted"
                    />
                  </span>
                    
                  <span>
                    <i
                      className="fa fa-thumbs-o-down text-muted"
                    />
                  </span>
                </small>
              </p>
            </div>
          </div>
        </div>
      </RecipeFavorite>
    </Router>
  </BrowserRouter>
</Provider>
`;
